src/components/checkbox/Checkbox.tsx

```diff
@@ -24,6 +24,8 @@ interface CheckboxProps extends Omit<BoxProps, 'onChange' | 'size'> {
   label?: string;
   /** Callback function that is fired when checkbox is toggled. */
   onChange?: (checked: boolean, event: ChangeEvent<HTMLInputElement>) => void;
+  /** Callback function that is fired when checkbox is clicked. */
+  onClick?: (event: MouseEvent) => void;
```

Comment:
What's the reason you now specifically add this `onClick`?

Reply:
Since our `Box` `[key:string]: unknown` indexation of props, Omitting any `BoxProps` we get all other props as `unknown`. So then `onClick` was unknown and I need to explicitely type it